hs_core/models.py

```diff
@@ -135,10 +135,19 @@ def can_view(self, request):
 def page_permissions_page_processor(request, page):
     cm = page.get_content_model()
     can_change_resource_flags = False
+    is_owner_user = False
+    is_edit_user = False
+    is_view_user = False
     if request.user.is_authenticated():
         if request.user.uaccess.can_change_resource_flags(cm):
             can_change_resource_flags = True
 
+        is_owner_user = cm.raccess.owners.filter(pk=request.user.pk).exists()
+        if not is_owner_user:
+            is_edit_user = cm.raccess.edit_users.filter(pk=request.user.pk).exists()
+            if not is_edit_user:
+                is_view_user = cm.raccess.view_users.filter(pk=request.user.pk).exists()
+
     owners = set(cm.raccess.owners.all())
     editors = set(cm.raccess.edit_users.all()) - owners
     viewers = set(cm.raccess.view_users.all()) - editors - owners
```

Comment:
None of these should be pass to `set()` causing them to be executed prematurely. In many views these aren't even needed, so it is wasteful to be consuming the entire query before its even clear that it is needed.